openedx/core/djangoapps/auth_exchange/tests/utils.py

```diff
@@ -86,16 +86,19 @@ def test_invalid_acess_token(self):
 
     def test_no_linked_user(self):
         UserSocialAuth.objects.all().delete()
+        Partial.objects.all().delete()
         self._setup_provider_response(success=True)
         self._assert_error(self.data, "invalid_grant", "access_token is not valid")
 
     def test_user_automatically_linked_by_email(self):
         UserSocialAuth.objects.all().delete()
+        Partial.objects.all().delete()
         self._setup_provider_response(success=True, email=self.user.email)
         self._assert_success(self.data, expected_scopes=[])
 
     def test_inactive_user_not_automatically_linked(self):
         UserSocialAuth.objects.all().delete()
+        Partial.objects.all().delete()
```

Comment:
@UmanShahzad @e-kolpakov This pattern seems weird to me too. I think that if you're subclassing the correct `django.test.TestCase` base class, then you shouldn't have to manually delete objects like this - the django base class should reset the database after each tests.